### ISSUE TITLE
The requested dns-domeneshop plugin does not appear to be installed

### ISSUE THREAD

**Issue by volklord** [domeneshop/certbot-dns-domeneshop#9]
Following the install and usage instructions in the readme file, certbot quits with error:

`The requested dns-domeneshop plugin does not appear to be installed`

Command I run:
`certbot certonly --authenticator dns-domeneshop -d '*.example.org'`

Command run to install:
`pip install certbot-dns-domeneshop`

----------------
```
pip --version
pip 22.2.2 from /usr/lib/python3.11/site-packages/pip (python 3.11)
```
----------------
```
certbot --version
certbot 2.1.0
```
----------------
OS: Fedora 37
----------------
```
/var/log/letsencrypt/letsencrypt.log
----------------
2023-03-11 18:07:11,393:DEBUG:certbot._internal.main:certbot version: 2.1.0
2023-03-11 18:07:11,394:DEBUG:certbot._internal.main:Location of certbot entry point: /usr/bin/certbot
2023-03-11 18:07:11,394:DEBUG:certbot._internal.main:Arguments: ['--authenticator', 'dns-domeneshop', '-d', '*.example.org']
2023-03-11 18:07:11,394:DEBUG:certbot._internal.main:Discovered plugins: PluginsRegistry(PluginEntryPoint#manual,PluginEntryPoint#nginx,PluginEntryPoint#null,PluginEntryPoint#standalone,PluginEntryPoint#webroot)
2023-03-11 18:07:11,410:DEBUG:certbot._internal.log:Root logging level set at 30
2023-03-11 18:07:11,410:DEBUG:certbot._internal.plugins.selection:Requested authenticator dns-domeneshop and installer None
2023-03-11 18:07:11,410:DEBUG:certbot._internal.plugins.selection:No candidate plugin
2023-03-11 18:07:11,411:DEBUG:certbot._internal.log:Exiting abnormally:
Traceback (most recent call last):
File "/usr/bin/certbot", line 8, in <module>
sys.exit(main())
^^^^^^
File "/usr/lib/python3.11/site-packages/certbot/main.py", line 19, in main
return internal_main.main(cli_args)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/lib/python3.11/site-packages/certbot/_internal/main.py", line 1736, in main
return config.func(config, plugins)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/lib/python3.11/site-packages/certbot/_internal/main.py", line 1571, in certonly
installer, auth = plug_sel.choose_configurator_plugins(config, plugins, "certonly")
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/lib/python3.11/site-packages/certbot/_internal/plugins/selection.py", line 255, in choose_configurator_plugins
diagnose_configurator_problem("authenticator", req_auth, plugins)
File "/usr/lib/python3.11/site-packages/certbot/_internal/plugins/selection.py", line 373, in diagnose_configurator_problem
raise errors.PluginSelectionError(msg)
certbot.errors.PluginSelectionError: The requested dns-domeneshop plugin does not appear to be installed
2023-03-11 18:07:11,412:ERROR:certbot._internal.log:The requested dns-domeneshop plugin does not appear to be installed
```

**Comment by volklord** [comment]
The problem seems to be related to (from)where certbot/dns-domeneshop is installed.
Installing certbot through the system package manager and dns-domeneshop through pip; makes certbot unable to find the plugin. On the other hand, if I install both with pip, then certbot finds the plugin, weird.
This wont work:
```txt
/usr/lib/python3.11/site-packages/certbot
/usr/local/lib/python3.11/site-packages/certbot_dns_domeneshop
```
This works:
```txt
/usr/local/lib/python3.11/site-packages/certbot
/usr/local/lib/python3.11/site-packages/certbot_dns_domeneshop
```
So, the question is why can't certbot find the plugin(s) if it isn't in the `same` directory? Is it restricting the paths on purpose?

**Comment by abisszett** [comment]
Related: https://community.letsencrypt.org/t/how-do-i-make-certbot-find-use-an-installed-plugin/198647
It seems not related to this plugin. Sorry for the noise and feel free to close the issue.

**Comment by volklord** [comment]
The simple workaround is to check the certbot executable `/usr/bin/certbot` doesn't include the `-s` in the shebang.
```sh
python3 --help
...
-s     : don't add user site directory to sys.path; also PYTHONNOUSERSITE
...
```
In my case, it was there and I just copied the file to `/usr/local/bin/certbot` and removed `-s`.
So, just in case someone else has same problem, replace `#! /usr/bin/python3 -sP` with `#! /usr/bin/python3 -P` and keep the rest. I don't get, why Fedora Linux or any other distro would put it in there.

Related:
https://community.letsencrypt.org/t/how-do-i-make-certbot-find-use-an-installed-plugin/198647
https://bugzilla.redhat.com/show_bug.cgi?id=2145166

It seems not related to this plugin. Sorry for the noise and feel free to close the issue.
EDIT: Added relevant links.
